examples/example-tutorial-todo-aip-app-sdk-2.x/src/useProjects.ts

```diff
@@ -1,19 +1,41 @@
 import { useCallback } from "react";
 import useSWR from "swr";
-import type { MockProject } from "./mocks";
 import Mocks from "./mocks";
 
+export interface IProject {
+  $apiName: string;
+  $primaryKey: string;
+  id: string;
+  name: string;
+  description: string;
+}
+
+export interface ITask {
+  $apiName: string;
+  $primaryKey: string;
+  id: string;
+  title: string;
+  description: string;
+  projectId: string;
+}
+
 function useProjects() {
-  const { data, isLoading, isValidating, error, mutate } = useSWR<
-    MockProject[]
-  >("projects", async () => {
-    // Try to implement this with the Ontology SDK!
-    return Mocks.getProjects();
-  });
+  const { data, isLoading, isValidating, error, mutate } = useSWR<IProject[]>(
+    "projects",
+    async () => {
+      // Try to implement this with the Ontology SDK!
+      const projectsList: IProject[] = (await Mocks.getProjects()).map((
+        project,
+      ) => ({
+        ...project,
+      }));
+      return projectsList;
+    },
+  );
```

Comment:
This feels like its redundant and could be done like this?

```suggestion
  const { data, isLoading, isValidating, error, mutate } = useSWR<IProject[]>(
    "projects",
    async () => {
      // Try to implement this with the Ontology SDK!
      const projectsList: IProject[] = await Mocks.getProjects();
      return projectsList;
    },
  );
```

Reply:
Right, but I thought it would be good to keep the mapping code here as when you implement the lesson you need a similar thing to map from your OSDK object to IProject.